all: simplify code identified by gosimple

Against afee90e311cd609197a37608e22aa2d983adcb16

```
$ gosimple ./...
graph/encoding/dot/dot_test.go:366:9: should use type conversion instead of struct literal (S1016)
graph/network/distance_test.go:162:3: should merge variable declaration with assignment on next line (S1021)
graph/network/distance_test.go:352:3: should merge variable declaration with assignment on next line (S1021)
graph/path/dijkstra_test.go:146:45: should omit comparison to bool constant, can be simplified to unique (S1002)
graph/path/floydwarshall_test.go:70:45: should omit comparison to bool constant, can be simplified to unique (S1002)
graph/path/johnson_apsp_test.go:70:45: should omit comparison to bool constant, can be simplified to unique (S1002)
lapack/testlapack/general.go:1138:3: should use 'return <expr>' instead of 'if <expr> { return <bool> }; return <bool>' (S1008)
lapack/testlapack/general.go:1198:3: should use 'return <expr>' instead of 'if <expr> { return <bool> }; return <bool>' (S1008)
stat/distuv/general_test.go:32:2: should use 'return <expr>' instead of 'if <expr> { return <bool> }; return <bool>' (S1008)
stat/distuv/laplace.go:108:2: redundant return statement (S1027)
stat/distuv/uniform.go:69:2: redundant return statement (S1027)
stat/samplemv/sample_test.go:50:9: should omit comparison to bool constant, can be simplified to !present[i][j] (S1002)
stat/samplemv/samplemv.go:114:4: should merge variable declaration with assignment on next line (S1021)
```

The code warned about in "graph/network/distance_test.go" is like that for stylistic reasons and will be left.

--- a/graph/encoding/dot/dot_test.go
+++ b/graph/encoding/dot/dot_test.go
@@ -363,7 +363,7 @@ type subGraph struct {
 
 func (g subGraph) ID() int { return g.id }
 func (g subGraph) Subgraph() graph.Graph {
-	return namedGraph{id: g.id, Graph: g.Graph}
+	return namedGraph(g)
 }
 
 func undirectedSubGraphFrom(g []intset, s map[int][]intset) graph.Graph {

--- a/graph/path/dijkstra_test.go
+++ b/graph/path/dijkstra_test.go
@@ -143,7 +143,7 @@ func TestDijkstraAllPaths(t *testing.T) {
 		}
 
 		np, weight, unique := pt.Between(test.NoPathFor.From(), test.NoPathFor.To())
-		if np != nil || !math.IsInf(weight, 1) || unique != false {
+		if np != nil || !math.IsInf(weight, 1) || unique {
 			t.Errorf("%q: unexpected path:\ngot: path=%v weight=%f unique=%t\nwant:path=<nil> weight=+Inf unique=false",
 				test.Name, np, weight, unique)
 		}

--- a/graph/path/floydwarshall_test.go
+++ b/graph/path/floydwarshall_test.go
@@ -67,7 +67,7 @@ func TestFloydWarshall(t *testing.T) {
 		}
 
 		np, weight, unique := pt.Between(test.NoPathFor.From(), test.NoPathFor.To())
-		if np != nil || !math.IsInf(weight, 1) || unique != false {
+		if np != nil || !math.IsInf(weight, 1) || unique {
 			t.Errorf("%q: unexpected path:\ngot: path=%v weight=%f unique=%t\nwant:path=<nil> weight=+Inf unique=false",
 				test.Name, np, weight, unique)
 		}

--- a/graph/path/johnson_apsp_test.go
+++ b/graph/path/johnson_apsp_test.go
@@ -67,7 +67,7 @@ func TestJohnsonAllPaths(t *testing.T) {
 		}
 
 		np, weight, unique := pt.Between(test.NoPathFor.From(), test.NoPathFor.To())
-		if np != nil || !math.IsInf(weight, 1) || unique != false {
+		if np != nil || !math.IsInf(weight, 1) || unique {
 			t.Errorf("%q: unexpected path:\ngot: path=%v weight=%f unique=%t\nwant:path=<nil> weight=+Inf unique=false",
 				test.Name, np, weight, unique)
 		}

--- a/lapack/testlapack/general.go
+++ b/lapack/testlapack/general.go
@@ -1135,10 +1135,9 @@ func isRightEigenvectorOf(a blas64.General, xRe, xIm []float64, lambda complex12
 		lambdax := make([]float64, n)
 		floats.AddScaled(lambdax, real(lambda), xRe)
 
-		if floats.Distance(xReAns, lambdax, math.Inf(1)) > tol {
-			return false
-		}
-		return true
+		// This is expressed as the inverse to catch the case
+		// xReAns_i = Inf and lambdax_i = Inf of the same sign.
+		return !(floats.Distance(xReAns, lambdax, math.Inf(1)) > tol)
 	}
 
 	// Complex eigenvector, and real or complex eigenvalue.
@@ -1195,10 +1194,9 @@ func isLeftEigenvectorOf(a blas64.General, yRe, yIm []float64, lambda complex128
 		lambday := make([]float64, n)
 		floats.AddScaled(lambday, real(lambda), yRe)
 
-		if floats.Distance(yReAns, lambday, math.Inf(1)) > tol {
-			return false
-		}
-		return true
+		// This is expressed as the inverse to catch the case
+		// yReAns_i = Inf and lambday_i = Inf of the same sign.
+		return !(floats.Distance(yReAns, lambday, math.Inf(1)) > tol)
 	}
 
 	// Complex eigenvector, and real or complex eigenvalue.

--- a/stat/distuv/general_test.go
+++ b/stat/distuv/general_test.go
@@ -29,10 +29,9 @@ type UniProbDist interface {
 }
 
 func absEq(a, b float64) bool {
-	if math.Abs(a-b) > 1e-14 {
-		return false
-	}
-	return true
+	// This is expressed as the inverse to catch the
+	// case a = Inf and b = Inf of the same sign.
+	return !(math.Abs(a-b) > 1e-14)
 }
 
 // TODO: Implement a better test for Quantile

--- a/stat/distuv/laplace.go
+++ b/stat/distuv/laplace.go
@@ -105,7 +105,6 @@ func (l Laplace) MarshalParameters(p []Parameter) {
 	p[0].Value = l.Mu
 	p[1].Name = "Scale"
 	p[1].Value = l.Scale
-	return
 }
 
 // Mean returns the mean of the probability distribution.

--- a/stat/distuv/uniform.go
+++ b/stat/distuv/uniform.go
@@ -66,7 +66,6 @@ func (u Uniform) MarshalParameters(p []Parameter) {
 	p[0].Value = u.Min
 	p[1].Name = "Max"
 	p[1].Value = u.Max
-	return
 }
 
 // Mean returns the mean of the probability distribution.

--- a/stat/samplemv/sample_test.go
+++ b/stat/samplemv/sample_test.go
@@ -47,7 +47,7 @@ func TestLatinHypercube(t *testing.T) {
 			allPresent := true
 			for i := 0; i < nSamples; i++ {
 				for j := 0; j < dim; j++ {
-					if present[i][j] == false {
+					if !present[i][j] {
 						allPresent = false
 					}
 				}

--- a/stat/samplemv/samplemv.go
+++ b/stat/samplemv/samplemv.go
@@ -111,9 +111,7 @@ func LatinHypercube(batch *mat.Dense, q distmv.Quantiler, src *rand.Rand) {
 	for i := 0; i < c; i++ {
 		p := perm(r)
 		for j := 0; j < r; j++ {
-			var v float64
-			v = f64()
-			v = v/r64 + float64(j)/r64
+			v := f64()/r64 + float64(j)/r64
 			batch.Set(p[j], i, v)
 		}
 	}

